addons/common/fnc_headDir.sqf

```diff
@@ -54,49 +54,35 @@ Author:
 #include "script_component.hpp"
 SCRIPT(headDir);
 
-private["_azimuth", "_angle", "_dif", "_infov", "_object", "_threed", "_do"];
-
 params ["_unit"];
-_object = param [1,_unit];
-_threed = false;
-
-_do = (typeName _object != typeName _unit);
-if (!_do) then {
-    _do = (_object != _unit);
-};
+private _object = param [1, _unit];
+private _isExternalCam = false;
 
-if (_do) then {
-    _angle = [(_unit call CBA_fnc_getPos),(_object call CBA_fnc_getPos)] call BIS_fnc_dirTo;
-} else {
-    _angle = 0;
-};
-if (_unit != player) then
-{
-    _azimuth = getdir _unit;
+private _azimuth = 0;
+if (_unit != player) then {
+    _azimuth = getDir _unit;
 } else {
-    private ["_position","_viewPos","_vector","_magnitude"];
-    _position = positionCameraToWorld [0, 0, 0];
-    _threed = if ((_position distance _unit)>2) then {true} else {false};
-    _viewPos = positionCameraToWorld [0, 0, 99999999];
-    _vector = [
-        (_viewPos select 0) - (_position select 0),
-        (_viewPos select 1) - (_position select 1),
-        (_viewPos select 2) - (_position select 2)
-    ];
-    _magnitude = [0, 0, 0] distance _vector;
-    _vector = [
-        (_vector select 0) / _magnitude,
-        (_vector select 1) / _magnitude,
-        (_vector select 2) / _magnitude
-    ];
+    private _camPos = positionCameraToWorld [0, 0, 0];
+    _isExternalCam = (_camPos distance _unit) > 2;
+    private _viewPos = positionCameraToWorld [0, 0, 99999999];

```

Comment:
Converting these to ASL would remove the terrain offset you get with the current function. Simply add `AGLToASL` before every `positionCameraToWorld`
...